File: examples/java/kora-java-scheduling-jdk/src/main/java/io/koraframework/example/scheduling/jdk/OnceScheduler.java
package io.koraframework.example.scheduling.jdk;

import java.time.temporal.ChronoUnit;
import io.koraframework.common.annotation.Component;
import io.koraframework.scheduling.jdk.annotation.ScheduleOnce;

@Component
public final class OnceScheduler {

    private int state = 0;

    @ScheduleOnce(delay = 50, unit = ChronoUnit.MILLIS)
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/main/kotlin/io/koraframework/kotlin/example/scheduling/jdk/OnceScheduler.kt
package io.koraframework.kotlin.example.scheduling.jdk

import io.koraframework.common.annotation.Component
import io.koraframework.scheduling.jdk.annotation.ScheduleAtFixedRate
import io.koraframework.scheduling.jdk.annotation.ScheduleOnce
import io.koraframework.scheduling.jdk.annotation.ScheduleWithFixedDelay
import java.time.temporal.ChronoUnit

@Component
class OnceScheduler {
    var state = 0
        private set

    @ScheduleOnce(delay = 50, unit = ChronoUnit.MILLIS)
    fun schedule() {
        state++
    }
}

